langchain/src/vectorstores/closevector/node.ts

```diff
@@ -0,0 +1,175 @@
+   * HNSW index, the arguments, and the document store from the directory,
+   * then creates a new HNSWLib instance with these values.
+   * @param directory The directory from which to load the vector store.
+   * @param embeddings The embeddings to be used by the CloseVectorNode instance.
+   * @returns A Promise that resolves to a new CloseVectorNode instance.
+   */
+  static async load(
+    directory: string,
+    embeddings: Embeddings,
+    credentials?: CloseVectorCredentials
+  ) {
+    const instance = await CloseVectorHNSWNode.load(directory, embeddings);
+    const vectorstore = new this(embeddings, instance.args, credentials);
+    return vectorstore;
+  }
+
+  /**
+   * Static method to create a new CloseVectorWeb instance from texts and metadata.
+   * It creates a new Document instance for each text and metadata, then
+   * calls the fromDocuments method to create the CloseVectorWeb instance.
+   * @param texts The texts to be used to create the documents.
+   * @param metadatas The metadata to be used to create the documents.
+   * @param embeddings The embeddings to be used by the CloseVectorWeb instance.
+   * @param args An optional configuration object for the CloseVectorWeb instance.
+   * @param credential An optional credential object for the CloseVector API.
+   * @returns A Promise that resolves to a new CloseVectorWeb instance.
+   */
+  static async fromTexts(
+    texts: string[],
+    metadatas: object[] | object,
+    embeddings: Embeddings,
+    args?: Record<string, unknown>,
+    credential?: CloseVectorCredentials
+  ): Promise<CloseVectorNode> {
+    const docs = CloseVector.textsToDocuments(texts, metadatas)
+    return await CloseVectorNode.fromDocuments(
+      docs,
+      embeddings,
+      args,
+      credential
+    );
+  }
+
+  /**
+   * Static method to create a new CloseVectorNode instance from documents. It
+   * creates a new CloseVectorNode instance, adds the documents to it, then returns
+   * the instance.
+   * @param docs The documents to be added to the HNSWLib instance.
+   * @param embeddings The embeddings to be used by the HNSWLib instance.
+   * @param args An optional configuration object for the HNSWLib instance.
+   * @param credentials An optional credential object for the CloseVector API.
+   * @returns A Promise that resolves to a new CloseVectorNode instance.
+   */
+  static async fromDocuments(
+    docs: Document[],
+    embeddings: Embeddings,
+    args?: Record<string, unknown>,
+    credentials?: CloseVectorCredentials
+  ): Promise<CloseVectorNode> {
+    const _args: Record<string, unknown> = args || {
+      space: "cosine",
+    };
+    const instance = new this(
+      embeddings,
+      _args as unknown as CloseVectorNodeArgs,
+      credentials
+    );
+    await instance.addDocuments(docs);
+    return instance;
+  }
+
+  static async imports(): Promise<{
+    HierarchicalNSW: typeof HierarchicalNSWT;
+  }> {
+    return await CloseVectorHNSWNode.imports();
```

Comment:
No need for return `await` here?

Also, I think no need for conditional import?

Reply:
@jacoblee93

True! I have removed "await" in the latest code.

If possible, I might prefer to reserve this method to allow developers to debug the inner instance and also align it with the closevector-web implementation, which can be used to preload the JS resources on the web.